### PR TITLE
hut: init at unstable-2022-03-02

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by papojari** [comment]
Checked everything. The only concern I have is with `meta.description`. All the packages of the package set `sourcehut` say *the sr.ht network*, no slashes or no *Sourcehut*.

**Comment by fgaz** [comment]
Those descriptions just mirror the READMEs, but sr.ht is actually the hosted service, while sourcehut is the software. sr.ht is often used as abbreviation though, so I included both to aid search

**Comment by papojari** [comment]
The `sourcehut` nix packages are also all sourcehut software and are not tied to `sr.ht, right? So I think your package should follow that. Anyone searching for `sourcehut` and not finding anything will probably also search for `sr.ht`.

**Comment by fgaz** [comment]
Yes but those are already named "sourcehut":

```
$ nix search nixpkgs sourcehut

* legacyPackages.x86_64-linux.sourcehut.buildsrht (0.75.2)
  Continuous integration service for the sr.ht network

* legacyPackages.x86_64-linux.sourcehut.coresrht (0.68.14)
  Core modules for sr.ht

...
```

So I'd like to keep both strings somewhere

**Comment by papojari** [comment]
> Yes but those are already named "sourcehut":

I didn't mean the package name but the description of this package.

**Comment by fgaz** [comment]
I know, I just want "sourcehut" to be _somewhere_ in a searchable string. `sourcehut.*` packages have it in the attribute path, but `hut` is just `hut`, so I have to put it in the description

**Comment by papojari** [comment]
Fine ;)

File: pkgs/applications/version-management/git-and-tools/hut/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  postBuild = ''
+    make $makeFlags completions doc/hut.1
+  '';
+
+  preInstall = ''
+    make $makeFlags install
+  '';
```
Comment (papojari) [inline pkgs/applications/version-management/git-and-tools/hut/default.nix:33]: Shouldn't this be put into the source repo?

Comment (fgaz) [reply pkgs/applications/version-management/git-and-tools/hut/default.nix:33]: what do you mean? there is nothing that I can send upstream as far as I can see

Comment (papojari) [reply pkgs/applications/version-management/git-and-tools/hut/default.nix:33]: Sry, I don't know what I had thought there. Commiting the generated manual page doesn't make sense.

Comment (papojari) [inline pkgs/applications/version-management/git-and-tools/hut/default.nix:33]: Isn't that unnescessary?

> The install phase is responsible for installing the package in the Nix store under out. The default installPhase creates the directory $out and calls make install.
> 
> https://nixos.org/manual/nixpkgs/stable/#ssec-install-phase

Comment (fgaz) [reply pkgs/applications/version-management/git-and-tools/hut/default.nix:33]: No, this is `buildGoModule`, not `mkDerivation`. `buildGoModule`'s `installPhase` simply copies the binaries, and does not call make, but here we want to install the completions and manual page too

**Review by auroraanna — APPROVED** [review]

File: pkgs/applications/version-management/git-and-tools/hut/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = with lib; {
+    homepage = "https://sr.ht/~emersion/hut/";
+    description = "A CLI tool for Sourcehut / sr.ht";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ fgaz ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/hut/default.nix:42]:
```suggestion
```

**Pull request merged**